onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -5649,12 +5649,87 @@ def aten_rnn_tanh_cell(
     raise NotImplementedError()
 
 
+@torch_op("aten::roll", trace_only=True)
 def aten_roll(
-    self: TensorType, shifts: Sequence[int], dims: Optional[Sequence[int]] = None
-) -> TensorType:
+    self: TTensor, shifts: Sequence[int], dims: Optional[Sequence[int]] = None
```

Comment:
It should be None. Because it can be integer or tuple.

Reply:
dims: Sequence[int] =() 

should be:

dims: Sequence[int] = ()

Reply:
Conclusion:
1. The shifts and dims are ```int``` or ```INT64```, no chance to be tuple.
2. For tuple type, we cannot handle it so far, because it will be converted to ```INT64 (tensor)```, but ```op.Concat(...axis=dim)``` need ```dim``` to be ```int```.
3. I'm high confident that ```roll(self, (1,0), (-1,1))``` (means shifts and dims are tuple) is really rare case. So we can skip/xfail it.